#Simple config file with just joysticks to use for the robot simulator
generic_hw_control_loop:
  loop_hz: 100
  cycle_time_error_threshold: 0.01
  ctre_mc_read_hz: 100
  pdp_read_hz: 20
  robot_iteration_hz: 20
  joystick_read_hz: 50
  match_data_read_hz: 2
  robot_controller_read_hz: 20


hardware_interface:
   joints:
       - {name: joystick0, type: joystick, id: 0, local: true}
       - {name: robot_code_ready_rio, local: true, type: ready}
       - {name: preshooter_limit_switch, type: digital_input, dio_channel: 9, invert: true, local: true}

joint_state_controller_rio:
   type:         joint_state_controller/JointStateController
   publish_rate: 100

match_state_controller:
   type:         match_state_controller/MatchStateController
   publish_rate: 20

robot_controller_state_controller:
   type: robot_controller_state_controller/RobotControllerStateController
   publish_rate: 20

joystick_controller:
    type: joystick_state_controller/JoystickStateController
    name: joystick0
    publish_rate: 50
